client/ayon_core/hosts/maya/plugins/publish/validate_animated_reference.py

```diff
@@ -16,7 +16,7 @@ class ValidateAnimatedReferenceRig(pyblish.api.InstancePlugin,
     hosts = ["maya"]
     families = ["animation.fbx"]
     label = "Animated Reference Rig"
-    accepted_controllers = ["transform", "locator"]
+    accepted_controllers = ["transform", "locator", "joint"]
```

Comment:
Is a `joint` a valid rig 'controller'? Is this correct? Does rig publishing itself even allow that?

This seems to hint that rig controls may only be transforms:
- https://github.com/ynput/ayon-core/blob/develop/client/ayon_core/hosts/maya/plugins/publish/collect_rig_sets.py#L20
- https://github.com/ynput/ayon-core/blob/develop/client/ayon_core/hosts/maya/plugins/publish/validate_rig_contents.py#L29

Why did this have to be added?

_Also, the PR description does not describe that this PR changes this - so if there's reasons to change this within this PR could you add it to the PR description?_

Reply:
> Is a `joint` a valid rig 'controller'? Is this correct? Does rig publishing itself even allow that?
> 
> This seems to hint that rig controls may only be transforms:
> 
> * https://github.com/ynput/ayon-core/blob/develop/client/ayon_core/hosts/maya/plugins/publish/collect_rig_sets.py#L20
> * https://github.com/ynput/ayon-core/blob/develop/client/ayon_core/hosts/maya/plugins/publish/validate_rig_contents.py#L29
> 
> Why did this have to be added?
> 
> _Also, the PR description does not describe that this PR changes this - so if there's reasons to change this within this PR could you add it to the PR description?_

I have added the description into the changelog. I know it sounds weird, but for some reason the validator errors out the joint when I added the joints into the set which is collected by the collector you mentioned.

Reply:
"For some reason" sounds like you're fixing something that you're unsure of whether it's intended behavior. To me the bigger question here is - why are joints part of the `controls` to begin with?

This seems like the wrong fix because the attribute naming just seems off - it's not `controls` it's validating? It seems to validate the `animated_skeleton` looking at the set members it's actually validating? If so, then the attribute called `validate_controls` should be `validate_skeleton` and the attribute `accepted_controllers` should maybe be `accepted_skeleton` (and potentially disallow transforms/locators if we do not want those as part of the skeleton?)

Reply:
> "For some reason" sounds like you're fixing something that you're unsure of whether it's intended behavior. To me the bigger question here is - why are joints part of the `controls` to begin with?
> 
> This seems like the wrong fix because the attribute naming just seems off - it's not `controls` it's validating? It seems to validate the `animated_skeleton` looking at the set members it's actually validating? If so, then the attribute called `validate_controls` should be `validate_skeleton` and the attribute `accepted_controllers` should maybe be `accepted_skeleton` (and potentially disallow transforms/locators if we do not want those as part of the skeleton?)

The code you showed is for rig family but not for animation family.(My bad I didn't notice you are showing rig set collector) If we double check on the collector `CollectFbxAnimation`, we are actually collecting everything. Maybe it's also good idea to fix the bug by implementing the ignore type as if we did in collect_animation if the constraint is being the reason to break the animation?

Reply:
What I'm saying that the attribute names are just invalid on this validator - it refers to "controllers" quite a bit but apparently validates something named `animated_skeleton` which sounds more like it is validating something that is expected to the be the skeleton only?

Then the next step is - if this is validating the skeleton. Then WHAT node types do we allow? I suppose we'd allow bones, of course - maybe parent groups? But do we actually allow locators? Etc.

Reply:
> What I'm saying that the attribute names are just invalid on this validator - it refers to "controllers" quite a bit but apparently validates something named `animated_skeleton` which sounds more like it is validating something that is expected to the be the skeleton only?
> 
> Then the next step is - if this is validating the skeleton. Then WHAT node types do we allow? I suppose we'd allow bones, of course - maybe parent groups? But do we actually allow locators? Etc.

we just keep it as it is without changing the original code of the validator, all we just need to change the collector to make sure it doesn't collect any maya-related constraint. About the naming of the `animated_skeleton`, the reason of naming `animated_skeleton` contributes to what the related sets have included, we can include set of control rigs inside the animated_skeleton set so that's why it is named like this. If it is not concise name, we can change it with different PRs or continue the discussion later

Reply:
Update: I guess I would just leave both validator and collector unchanged as the changes can affect the pipeline between fbx animation from maya to unreal.